### ISSUE TITLE
Unhandled type-error when running `describe-slot`

### ISSUE THREAD

**Issue by Lithrein** [atlas-engineer/nyxt#1226]
**Describe the bug**

Describe slot crashes nyxt

**Precise recipe to reproduce the issue**

Compile from master, launch nyxt with `nyxt -I` and run `describe-slot`.

**Information**
- How you installed Nyxt (Guix pack, package manager, build from source): from source
- Information from command copy-system-information:
```
Nyxt version: 2-pre-release-5-970-g42fc020e 
Operating system kernel: Linux 5.4.0-65-generic
Lisp implementation: SBCL 2.0.1.debian
Features: (WEBKIT2 WEBKIT2-2.30 WEBKIT2-TRACKING WEBKIT2-MUTE WEBKIT2-EMOJI
           WEBKIT2-MEDIA WEBKIT2-SANDBOXING GTK-3-22 GTK-3-20 GTK-3-18 GTK-3-16
           GTK-3-14 GTK-3-12 GTK-3-10 GTK-3-8 GTK-3-6 GTK-3-4 GTK GDK-3-22
           GDK-3-20 GDK-3-18 GDK-3-16 GDK-3-14 GDK-3-12 GDK-3-10 GDK-3-8
           GDK-3-6 GDK-3-4 CAIRO-1-10 CAIRO-1-12 GDK-PIXBUF GLIB-2-30 GLIB-2-32
           GLIB-2-34 GLIB-2-36 GLIB-2-38 GLIB-2-40 GLIB-2-42 GLIB-2-44
           GLIB-2-46 GLIB-2-48 GLIB-2-50 GLIB-2-52 GLIB-2-54 GLIB-2-56
           GLIB-2-58 GLIB FSET-EXT-STRINGS CUSTOM-HASH-TABLE-NATIVE SWANK
           PLUMP-UTF-32 GLOBAL-VARS DECLARE-TYPES PARENSCRIPT NAMED-READTABLES
           21BIT-CHARS CHUNGA FLEXI-STREAMS CLOSER-MOP CL-PPCRE-UNICODE
           CL-UNICODE CL-PPCRE BORDEAUX-THREADS SPLIT-SEQUENCE GRAY-STREAMS
           FLAT-NAMESPACE X86-64 UNIX CFFI FLAT-NAMESPACE SEQUENCE-EMPTYP
           FAST-IO-SV FAST-IO SBCL-USES-SB-ROTATE-BYTE ASDF-SYSTEM-CONNECTIONS
           CL-JSON-CLOS CL-JSON THREAD-SUPPORT QUICKLISP
           SB-BSD-SOCKETS-ADDRINFO ASDF3.3 ASDF3.2 ASDF3.1 ASDF3 ASDF2 ASDF
           OS-UNIX NON-BASE-CHARS-EXIST-P ASDF-UNICODE X86-64 64-BIT ANSI-CL
           COMMON-LISP ELF GENCGC IEEE-FLOATING-POINT LINUX LITTLE-ENDIAN
           PACKAGE-LOCAL-NICKNAMES SB-CORE-COMPRESSION SB-LDB SB-PACKAGE-LOCKS
           SB-THREAD SB-UNICODE SBCL UNIX)
```

**Output when started from a shell**

```
Unhandled TYPE-ERROR in thread #<SB-THREAD:THREAD "Anonymous thread" RUNNING
                                  {10064DC193}>:
  The value
    (PASSWORD:EXECUTABLE PASSWORD:USER-PASSWORD-STORE-INTERFACE)
  is not of type
    (OR NULL (CONS STRING T))
  when binding STR::STRINGS

Backtrace for: #<SB-THREAD:THREAD "Anonymous thread" RUNNING {10064DC193}>
0: (STR:JOIN " " (PASSWORD:EXECUTABLE PASSWORD:USER-PASSWORD-STORE-INTERFACE)) [external]
1: (PROMPTER::FORMAT-PROPERTIES (:NAME PASSWORD:EXECUTABLE :CLASS PASSWORD:USER-PASSWORD-STORE-INTERFACE) NIL)
2: ((:METHOD INITIALIZE-INSTANCE :AFTER (PROMPTER:SUGGESTION)) #<PROMPTER:SUGGESTION {10064E00F3}> :SOURCE #<NYXT::SLOT-SOURCE {1008BB1003}> :INPUT "") [fast-method]
3: ((LAMBDA (SB-PCL::|.P0.| SB-PCL::|.P1.| SB-PCL::|.P2.| SB-PCL::|.P3.|)) #<unavailable argument> #<unavailable argument> #<unavailable argument> #<unavailable argument>)
4: ((LAMBDA NIL :IN PROMPTER::UPDATE))
5: ((LAMBDA NIL :IN BORDEAUX-THREADS::BINDING-DEFAULT-SPECIALS))
6: ((FLET SB-UNIX::BODY :IN SB-THREAD::NEW-LISP-THREAD-TRAMPOLINE))
7: ((FLET "WITHOUT-INTERRUPTS-BODY-4" :IN SB-THREAD::NEW-LISP-THREAD-TRAMPOLINE))
8: ((FLET SB-THREAD::WITH-MUTEX-THUNK :IN SB-THREAD::NEW-LISP-THREAD-TRAMPOLINE))
9: ((FLET "WITHOUT-INTERRUPTS-BODY-1" :IN SB-THREAD::CALL-WITH-MUTEX))
10: (SB-THREAD::CALL-WITH-MUTEX #<CLOSURE (FLET SB-THREAD::WITH-MUTEX-THUNK :IN SB-THREAD::NEW-LISP-THREAD-TRAMPOLINE) {7FB10FDAED7B}> #<SB-THREAD:MUTEX "thread result lock" owner: #<SB-THREAD:THREAD "Anonymous thread" RUNNING {10064DC193}>> NIL T NIL)
11: (SB-THREAD::NEW-LISP-THREAD-TRAMPOLINE #<SB-THREAD:THREAD "Anonymous thread" RUNNING {10064DC193}> NIL #<CLOSURE (LAMBDA NIL :IN BORDEAUX-THREADS::BINDING-DEFAULT-SPECIALS) {10064DC13B}> NIL)
12: ("foreign function: call_into_lisp")
13: ("foreign function: new_thread_trampoline")
```

**Comment by Ambrevar** [comment]
I believe this is the same as #1217.  Can you try on master again and confirm it is fixed?

**Comment by Lithrein** [comment]
I confirm this is the same issue, and it is now fixed. This issue can be closed and merged with #1217.

 (However, the new harmonization in Makefile (--eval '(asdf:load-asd "nyxt.asd")' instead of --load nyxt.asd) and nyxt.asd (no more colons before symbols) causes compiles errors, I had to use the old Makefile. That is either a mistake on my part (old version of sbcl?) or it should probably be another issue).
